src/Database/PDO.php

```
@@ -41,7 +42,26 @@ public function __construct(
      */
     public function __call(string $method, array $args): mixed
     {
-        return $this->pdo->{$method}(...$args);
+        try {
+            return $this->pdo->{$method}(...$args);
+        } catch (\Throwable $e) {
+            if (Connection::hasError($e)) {
+                Console::warning('[Database] Lost connection detected. Reconnecting...');
+
+                $inTransaction = $this->pdo->inTransaction();
+
+                // Attempt to reconnect
+                $this->reconnect();
+
+                // If we weren't in a transaction, also retry the query
+                // In a transaction we can't retry as the state is attached to the previous connection
+                if (!$inTransaction) {
```

If transaction === true  Do we want to throw a specific error ? 
otherwise return $this->pdo->{$method}(...$args); ?

---

If it was a connection error, but we were in a transaction, we'll throw the original exception